modules/src/core/ics02_client/client_state.rs

```diff
@@ -11,6 +11,7 @@ use crate::clients::ics07_tendermint::client_state;
 use crate::core::ics02_client::client_type::ClientType;
 use crate::core::ics02_client::error::Error;
 use crate::core::ics02_client::trust_threshold::TrustThreshold;
+use crate::core::ics23_commitment::specs::ProofSpecs;
```

Comment:
I don't think we can have a `proof_spec()` method in `ClientState` because that is Tendermint specific, and it should be possible to create a client that doesn't use ICS23 style proofs. By extension, verification functions that rely on proof specs are also client-specific so IMHO they should not be present in the default implementation or trait API.